### ISSUE TITLE
Detections for nuScenes mini dataset

### ISSUE THREAD

**Issue by tiffanyyk** [aleksandrkim61/EagerMOT#9]
Thanks for sharing your code! 
I'm wondering if the CenterPoint and MMDetection detections are provided for the v1.0-mini version of the nuScenes dataset. As mentioned in the README, I changed `SPLIT` to mini-train / mini-val, but the CenterPoint and MMDetection folders from the [google drive link](https://drive.google.com/drive/folders/1MpAa9YErhAZNEJjIrC4Ky21YfNj2jatM) contain detections for the (full) validation and test sets only. Am I missing something here? Thanks in advance!

**Comment by aleksandrkim61** [comment]
Hi!

1) On NuScenes, if I remember correctly, the mini-train/mini-val are just subsets of the train/val splits, so the detections are exactly the same. To avoid duplicating data on disk, I simply renamed my val/test folders to mini-val/mini-test when running early experiments and then renamed them back when running experiments on full data. Alternatively, you can also just duplicate files.

2) The detections I extracted myself (not given by other implementations) are not provided for the training set - I never had them because the tracking framework just took detections as-is, without trying to train detector models. 
If you need training set detections, you might try to find them in other implementations. Alternatively, the framework should work well with any detections of comparable quality, so maybe try something you already have?

For CenterPoint, look at the original [model zoo](https://github.com/tianweiy/CenterPoint/blob/master/configs/nusc/README.md), where I got my detections for the `centerpoint_voxel_1440_dcn(flip)` configuration. Maybe some other model config has training results.
For MMDetection, I used their model Cascade Mask R-CNN | X-101_32x4d with this config from the [model zoo](https://github.com/open-mmlab/mmdetection3d/tree/master/configs/nuimages):
```py
config_file = 'configs/nuimages/cascade_mask_rcnn_x101_32x4d_fpn_1x_nuim.py'
checkpoint_file = '../checkpoints/cascade_mask_rcnn_x101_32x4d_fpn_1x_nuim_20200906_134611-47db31b0.pth'
model_name = "cascade_mask_rcnn_x101"
```

Hope this helps!

EDIT: updated MMDetection link

**Comment by tiffanyyk** [comment]
Thanks for the helpful clarifications! Closing this issue now.

**Comment by nmll** [comment]
May I ask about where is configs/nuimages/cascade_mask_rcnn_x101_32x4d_fpn_1x_nuim.py？

**Comment by aleksandrkim61** [comment]
Hi, it looks like I linked to the wrong model zoo (mmdetection instead of mmdetection3d).

The link for 2D detection models from MMDetection is this https://github.com/open-mmlab/mmdetection3d/tree/master/configs/nuimages.
There, you will find the configs and the models available. The config I specified in the previous reply is correct (for Cascade Mask R-CNN | X-101_32x4d), but looks like the checkpoint and performance has improved. If you would like to use the exact checkpoint I have used, you will find it in the commit from September 8. 
P.S. The config is not important if you just want to run their pre-trained model, so it is only there to point to the correct model.

**Comment by nmll** [comment]
ok，Thanks a lot!

**Comment by gayathri81** [comment]
Hello, one quick question on the 2d detections results.
I have v1.0-mini version of the nuScenes dataset and centerpoint detections for the centerpoint_voxel_1440_dcn(flip) from the link you provided. To get the 2d detection can I run the Cascade Mask R-CNN | X-101_32x4d  model on the nuscenes data to get the 2d detections and plug these 3d and 2d to the tracking? Please correct me if Im wrong here.

**Comment by gayathri81** [comment]
@tiffanyyk  Can you please help me in understanding how to relate the detections downloaded from the drive to the images in the nuscenes mini version. Appreciate your help.
